### PR TITLE
chore(utils): Add helper component for transition-group

### PR DESCRIPTION
**Pull request by mayank99**
Extracted out common transition-group logic from ExpandableBlock and TableRow into new `WithCSSTransition` component. This helper component may be used in other places, like the Sidenav submenu.

## Checklist

- [ ] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [x] ~~Add component features demo in Storybook (different stories)~~
- [x] ~~Approve test images for new stories~~
- [x] ~~Add screenshots of the key elements of the component~~

### CODE REVIEW

File: src/core/utils/components/WithCSSTransition.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { CSSTransition } from 'react-transition-group';
+import { CSSTransitionProps } from 'react-transition-group/CSSTransition';
+
+export const WithCSSTransition = (
```
Comment (bentleyvk) [inline src/core/utils/components/WithCSSTransition.tsx:9]: Is it possible to have some tests?

File: src/core/utils/components/WithCSSTransition.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  props: Partial<CSSTransitionProps> & {
+    children: JSX.Element;
+    orientation?: 'vertical' | 'horizontal';
+  },
+) => {
+  const { in: visible, orientation = 'vertical', children, ...rest } = props;
```
[the remark this answers is not in the document]
Comment (mayank99) [reply src/core/utils/components/WithCSSTransition.tsx:15]: `in` is a keyword so it's forbidden

File: src/core/utils/components/WithCSSTransition.tsx (hunk 1 above)
Comment (mayank99) [reply src/core/utils/components/WithCSSTransition.tsx:9]: i have absolutely no clue how to test this

**Review by bentleyvk — APPROVED** [review]

File: src/core/utils/components/WithCSSTransition.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      unmountOnExit={true}
+      onEnter={(node) => {
+        node.style[`min${dimensionCamelCase}`] = 'initial';
+        node.style[dimension] = '0px';
+      }}
+      onEntering={(node) => {
+        node.style[dimension] = `${expandedSize.current}px`;
+      }}
+      onEntered={(node) => {
+        node.style[`min${dimensionCamelCase}`] = '';
```
Comment (bentleyvk) [inline src/core/utils/components/WithCSSTransition.tsx:34]: Why you need this?

**Review by gretanausedaite — APPROVED** [review]

**Pull request merged**